crates/tests/mqtt_tests/src/test_mqtt_server.rs

```diff
@@ -86,69 +89,76 @@ fn spawn_broker(port: u16) {
     });
 
     std::thread::spawn(move || {
-        let mut rx = tx.connect(200).unwrap();
         tx.subscribe("#").unwrap();
 
         loop {
-            if let Some(message) = rx.recv().unwrap() {
-                for chunk in message.payload.into_iter() {
-                    let mut bytes: Vec<u8> = vec![];
-                    for byte in chunk.into_iter() {
-                        bytes.push(byte);
-                    }
-                    let payload = match std::str::from_utf8(bytes.as_ref()) {
+            let notification = match rx.recv().unwrap() {
+                Some(v) => v,
+                None => continue,
```

Comment:
Is it a good idea to continue on `None`? A `None` is typically returned when the broker closes the channel from the other end, right? Then what's the point in continuing? If it's still valid, better add a comment mentioning why.

Reply:
From the rumqttd source code comments, I can understand that there can be a situation when there are no messages in the queue, then it returns None. But after some time there can be a newer message as well. So, I feel it's better to continue. 

Reply:
This is coming from this [example](https://github.com/bytebeamio/rumqtt/blob/2e0eafc41482f4309bf9e5a6f28de45a84e681a5/rumqttd/examples/singlenode.rs)